SimG4Core/SensitiveDetector/interface/SensitiveDetector.h

```diff
@@ -22,32 +21,29 @@ class DDCompactView;
 class SensitiveDetector : public G4VSensitiveDetector
 {
 public:
-  explicit SensitiveDetector(std::string & iname, const DDCompactView & cpv,
+  explicit SensitiveDetector(const std::string & iname, const DDCompactView & cpv,
 			     const SensitiveDetectorCatalog & ,
 			     edm::ParameterSet const & p);
-  virtual ~SensitiveDetector();
-  virtual void Initialize(G4HCofThisEvent * eventHC);
+  ~SensitiveDetector() override;
+  void Initialize(G4HCofThisEvent * eventHC) override;
   virtual void clearHits() = 0;
-  virtual G4bool ProcessHits(G4Step * step ,G4TouchableHistory * tHistory) = 0;
+  G4bool ProcessHits(G4Step * step ,G4TouchableHistory * tHistory) override = 0;
   virtual uint32_t setDetUnitId(G4Step * step) = 0;
   void Register();
-  virtual void AssignSD(const std::string & vname);
-  virtual void EndOfEvent(G4HCofThisEvent * eventHC); 
+  void AssignSD(const std::string & vname);
+  void EndOfEvent(G4HCofThisEvent * eventHC) override; 
+
   enum coordinates {WorldCoordinates, LocalCoordinates};
-  Local3DPoint InitialStepPosition(G4Step * step, coordinates);
-  Local3DPoint FinalStepPosition(G4Step * step, coordinates);
+  Local3DPoint InitialStepPosition(const G4Step * step, coordinates);
+  Local3DPoint FinalStepPosition(const G4Step * step, coordinates);
+
   inline Local3DPoint ConvertToLocal3DPoint(const G4ThreeVector& point)
   {
     Local3DPoint res(point.x(),point.y(),point.z());
     return std::move(res);
```

Comment:
do not understand where this idiom comes from: it is clearly stated a bit everywhere that one shall NOT move return values. it usually prevent RVO that is more efficient...